src/vm/builtins/Array.zig

```diff
@@ -60,9 +61,34 @@ fn array_reverse(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value
     return Code.Value.Object.Array.init(vm.allocator, new_array) catch Code.Value{ .none = {} };
 }
 
-fn length(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value {
-    _ = vm;
+fn foreach(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value {
+    if (!((arguments[0] == .object and (arguments[0].object == .array or arguments[0].object == .map)) and (arguments[1] == .object and arguments[1].object == .function))) {
+        return Code.Value{ .none = {} };
+    }
+
+    const callback = arguments[1].object.function;
+
+    if (arguments[0].object == .map) {
+        const map = arguments[0].object.map;
```

Comment:
Use a switch statement in here, `switch (arguments[0]) {}` and leave else cases empty until the end you just return none, check `length` function for an example